stable/datadog/templates/container-trace-agent.yaml

```diff
@@ -37,6 +37,10 @@
       mountPath: {{ template "datadog.confPath" . }}/datadog.yaml
       subPath: datadog.yaml
     {{- end }}
+    {{- if .Values.datadog.dogstatsd.useSocketVolume }}
+    - name: dsdsocket
+      mountPath: "/var/run/datadog"
+    {{- end }}
```

Comment:
This is the socket used by the dogstatsd server. 

I think it makes sens to add the new parameters `.Values.datadog.apm.useSocketVolume|sockerPath` 
```
datadog:
  apm:
    ## @param useSocketVolume - boolean - optional
    ## Enable APM over Unix Domain Socket
    ## ref: https://docs.datadoghq.com/agent/kubernetes/apm
    #
    # useSocketVolume: true

    ## @param socketPath - string - optional
    ## Path to the Trace-Agent socket
    #
    # socketPath: /var/run/datadog/apm.socket

The EnvVar `DD_APM_RECEIVER_SOCKET ` can also be added to the `trace-agent` container when `.Values.datadog.apm.useSocketVolume` is `true`.

WDYT?

Reply:
That sounds good to me.  I noticed however that the helm chart hard codes `/var/run/datadog` as the mount.  If someone used `/var/datadog/dsd` as the path instead, I don't think the mount part would work right.  Maybe also document this limitation?

Reply:
yes. it requires to have a new parameter for allowing the overwrite the hostpath. If you feel confortable you can also add this in your PR.

Also I think it makes sense to add a new "Volume.hostPath" for the trace-agent like it is done here for dogstatsd:
https://github.com/helm/charts/blob/master/stable/datadog/templates/daemonset-volumes-linux.yaml#L8